### PR TITLE
[react-redux] Simplify connect return to work stricter generic checks of TypeScript 2.4

### PR DESCRIPTION
**Pull request by codecorsair**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/DefinitelyTyped/DefinitelyTyped/issues/17829>
- [x] Increase the version number in the header if appropriate.

### CODE REVIEW

File: types/react-redux/index.d.ts [hunk 1]
```diff
@@ -19,16 +19,8 @@ export interface DispatchProp<S> {
...
-}
-
-interface ComponentDecoratorInfer<TMergedProps> {
-    <T>(component: Component<T & TMergedProps>): ComponentClass<T>;
-}
-
-interface ComponentMergeDecorator<TMergedProps, TOwnProps> {
-    (component: Component<TMergedProps>): ComponentClass<TOwnProps>;
+export interface WrapWithDispatch {
+    <P, TComponent extends Component<P>>(component: TComponent): TComponent;
```
Comment (thasner) [inline types/react-redux/index.d.ts]: Does this lose the default behavior of the returned component being `ComponentClass<TOwnProps>`?

Comment (codecorsair) [inline types/react-redux/index.d.ts]: The returned component is typed the same as the component passed in to this method as if nothing was changed on the original component. Which is any object that extends `ComponentClass<P> | StatelessComponent<P>`.

**Comment by codecorsair** [comment]
Did I miss something that needed to be changed elsewhere with changing the version comment to make Travis build this file successfully?

**Comment by RyanCavanaugh** [comment]
@codecorsair Please fix the failures indicated in the Travis CI log.

**Pull request closed without merging**